modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/UpdateContentCommand.java

```diff
@@ -206,8 +205,8 @@ private Attachments mergeExistingAndUpdatedAttachments( final Attachments origin
         if ( !params.isClearAttachments() )
         {
             originalAttachments.stream().forEach( a -> attachments.put( a.getBinaryReference(), a ) );
+            params.getRemoveAttachments().stream().forEach( attachments::remove );
```

Comment:
why it worked before?

Reply:
`attachments` are empty outside of this block, there is no reason to remove anything from them